apps/pipelines/models.py

```diff
@@ -67,12 +68,43 @@ def version_display(self) -> str:
             return ""
         return f"v{self.version_number}"
 
+    @classmethod
+    def create_default(cls, team):
+        from apps.pipelines.nodes.nodes import EndNode, StartNode
+
+        default_name = "New Pipeline"
+        existing_pipeline_count = cls.objects.filter(team=team, name__startswith=default_name).count()
+
+        start_node = {
+            "id": str(uuid4()),
+            "type": "pipelineNode",
+            "position": {
+                "x": -200,
+                "y": 200,
+            },
+            "data": {"id": str(uuid4()), "type": StartNode.__name__},
+        }
+        end_node = {
+            "id": str(uuid4()),
+            "type": "pipelineNode",
+            "position": {"x": 1000, "y": 200},
+            "data": {"id": str(uuid4()), "type": EndNode.__name__},
+        }
+        default_nodes = [start_node, end_node]
+        new_pipeline = cls.objects.create(
+            team=team,
+            data={"nodes": default_nodes, "edges": []},
+            name=f"New Pipeline {existing_pipeline_count + 1}",
+        )
+        new_pipeline.set_nodes()
+        return new_pipeline
+
     def get_absolute_url(self):
         return reverse("pipelines:details", args=[self.team.slug, self.id])
 
-    def set_nodes(self, nodes: list[FlowNode]) -> None:
+    def set_nodes(self) -> None:
```

Comment:
nit: what about renaming this to "update_nodes_from_data"

Reply:
https://github.com/dimagi/open-chat-studio/pull/929/commits/de8400045bcc58a2ae821e28e8ee296a179cb991